pkg/v2/tkr/webhook/cluster/tkr-resolver/cluster/cluster.go

```diff
@@ -122,47 +111,20 @@ func (cw *Webhook) constructQuery(cluster *clusterv1.Cluster, clusterClass *clus
 		return nil, err
 	}
 
-	cpQuery := cw.filterOSImageQuery(tkr, tkrData, &data.OSImageQuery{
-		K8sVersionPrefix: cluster.Spec.Topology.Version,
-		TKRSelector:      tkrSelector,
-		OSImageSelector:  osImageSelector,
-	})
-
-	if cluster.Spec.Topology.Workers == nil {
-		return &data.Query{ControlPlane: cpQuery}, nil
-	}
-
-	mdQueries, err := cw.constructMDOSImageQueries(tkr, cluster, clusterClass, tkrSelector)
+	query, err := resolution.ConstructQuery(cluster.Spec.Topology.Version, cluster, clusterClass)
```

Comment:
**For reviewers:**

Most of the logic from `cw.constructQuery()` is moved into `resolution.ConstructQuery()`